src/main/java/ogc/rs/processes/ProcessesRunnerImpl.java

```diff
@@ -30,18 +32,33 @@ public class ProcessesRunnerImpl implements ProcessesRunnerService {
   private final UtilClass utilClass;
   private final JsonObject config;
   private final Vertx vertx;
-  private final DataFromS3 dataFromS3;
+  private String S3_BUCKET;
+  private String S3_REGION;
+  private String S3_ACCESS_KEY;
+  private String S3_SECRET_KEY;
+  private HttpClientOptions httpClientOptions;
+  private HttpClient httpClient;
   Logger LOGGER = LogManager.getLogger(ProcessesRunnerImpl.class);
 
-  ProcessesRunnerImpl(PgPool pgPool, WebClient webClient, JsonObject config, DataFromS3 dataFromS3, Vertx vertx) {
+  ProcessesRunnerImpl(PgPool pgPool, WebClient webClient, JsonObject config, Vertx vertx) {
     this.pgPool = pgPool;
     this.webClient = webClient;
     this.utilClass = new UtilClass(pgPool);
     this.config = config;
-    this.dataFromS3=dataFromS3;
     this.vertx=vertx;
   }
-
+  private DataFromS3 getS3Object(JsonObject config){
+    S3_BUCKET = config.getString("s3BucketUrl");
+    S3_REGION = config.getString("awsRegion");
+    S3_ACCESS_KEY = config.getString("awsAccessKey");
+    S3_SECRET_KEY = config.getString("awsSecretKey");
+
+    httpClientOptions = new HttpClientOptions().setSsl(true);
```

Comment:
You might need to include this here when making the `HttpClient` in order to allow for testing:
https://github.com/datakaveri/ogc-resource-server/blob/main/src/main/java/ogc/rs/apiserver/ApiServerVerticle.java#L157-L160